Content.Server/Chat/LastMessageBeforeDeath.cs

```diff
@@ -46,58 +56,87 @@ public async void AddMessage(string playerName, EntityUid player, string message
             }
 
-            // Replace the existing message with the new one
-            _playerData[player][playerName] = message;
-
+            var playerMessages = _playerData[player] as OrderedDictionary ?? throw new InvalidOperationException("Player messages dictionary is not initialized.");
+            playerMessages[playerName] = formattedMessage;
         }
 
-
-        // Send the messages through the webhook.
+        // Send messages through the webhook on round end
         public async void OnRoundEnd(WebhookIdentifier? webhookId)
         {
             if (!webhookId.HasValue)
                 return;
 
             var id = webhookId.Value;
-
             StringBuilder allMessagesString = new StringBuilder();
-            foreach (var player in _playerData)
+            foreach (DictionaryEntry player in _playerData)
             {
-                var playerUid = player.Key;
+                var playerUid = (EntityUid)player.Key;
                 if (_mobStateSystem.IsDead(playerUid))
                 {
-                    foreach (var playerName in player.Value)
+                    var playerMessages = player.Value as OrderedDictionary;
+                    if (playerMessages != null)
                     {
-                        allMessagesString.AppendLine($"{playerName.Key}: {playerName.Value}");
+                        foreach (DictionaryEntry playerName in playerMessages)
+                        {
+                            allMessagesString.AppendLine($"{playerName.Value}");
+                        }
                     }
                 }
             }
+
             if (allMessagesString.ToString().Length == 0)
             {
                 allMessagesString.AppendLine($"No messages found.");
             }
 
-            var payload = new WebhookPayload
-            {
-                Content = allMessagesString.ToString()
-            };
+            var messages = SplitMessage(allMessagesString.ToString(), 2000);
             var discordWebhook = new DiscordWebhook();
-            var response = await discordWebhook.CreateMessage(id, payload);
-
-            //if (response.IsSuccessStatusCode)
-            //{
-            //    _sawmill.Info("Last Messages Before Death sent successfully.");
-            //}
-            //else
-            //{
-            //    _sawmill.Warning($"Failed to send last messages before death. Status code: {response.StatusCode}");
-            //}
+            int messageCount = 0;
+            foreach (var message in messages)
+            {
+                if (messageCount >= 30)
+                {
+                    await Task.Delay(60000); // Wait for 1 minute
+                    messageCount = 0;
+                }
+                var payload = new WebhookPayload { Content = message };
+                var response = await discordWebhook.CreateMessage(id, payload);
+                messageCount++;
```

Comment:
Yeah... 
I've added the delay of 2 seconds between messages. 
I have also decreased the maximum length of the IC message itself, so if it breaches the maximum, it will be randomly cut off below the maximum size.